FindVitis.cmake

```diff
@@ -57,81 +57,106 @@
 include(FindPackageHandleStandardArgs)
 
 #if Xilinx tools correctly installed they are added to $ENV{PATH} one of CMake's default search paths
+list(LENGTH Vitis_FIND_COMPONENTS componentsSize)
+if(componentsSize EQUAL 0)
+	message(STATUS "Error: AIE version needed as COMPONENT, specifiy AIE or AIE-ML")
+elseif(componentsSize GREATER 1)
+	message(STATUS "Error: only a single AIE version supported as COMPONENT, either specifiy AIE or specify AIE-ML")
+else()
```

Comment:
I think having a default of AIE when no components are specified is a good thing.  I also think that we need to allow multiple architectures to be found simultaneously (see my proposed patch).  I was also using "AIE2" since that it what is used in the single core compiler, rather than AIE-ML.  (But I wish there was more consistent terminology here.)

Reply:
In particular, I'd like to have aiecc be able to target any device, selected on the command line.  see https://github.com/Xilinx/mlir-aie/pull/367/commits/2e36894fbfb5ff6168b1c3eacd5656b464027ba1

Reply:
@stephenneuendorffer I added support for multiple components and set AIE as default in case the user does not specify the COMPONENTS keyword.
AIE2 used as name instead of AIE-ML